src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -434,6 +443,15 @@ private static string GenerateUpgradeReport(NuGetProject nuGetProject, string ba
 
                     uiService.EndOperation();
 
+                    // don't show "Succeeded" if we actually cancelled...
+                    if ((!continueAfterPreview) || (!acceptedLicense))
+                    {
+                        if (status == NuGetOperationStatus.Succeeded)
```

Comment:
For which scenario is this check needed?

Reply:
The scenario here is where either of the two specific cancel flags were encountered but the operation status was still "Succeeded" for some reason by the time the code hit the finally block.  The finally block is where the telemetry is logged, and the control flow can jump there at any time.  I observed this in my testing.

So the point of this check is that if we arrive in the finally block with one or both of the known cancellation scenarios tripped, but the operation status is still "success", we know that something went wrong and we should mark the status as cancelled...

Reply:
The status is set at the top, so I guess my gut instinct is, don't set it at the top? 

Is there a case, where the above conditions are not but the status is not Succeeded?